### ISSUE TITLE
Large Ram Usage

### ISSUE THREAD

**Issue by DeathTBO** [Codeusa/Borderless-Gaming#108]
It starts at 20mb then gradually rises. The largest I've seen was 160mb.

**Comment by psouza4** [comment]
I've had it running for weeks and it's sitting at 14.6 MB.  I would check your system for viruses that may be attaching themselves to host processes.

**Comment by DeathTBO** [comment]
Currently scanning for viruses. I have a second machine with the same specs, and it too has the problem. I recently did a scan on the second machine and there were no viruses.

**Comment by psouza4** [comment]
You may also want to experiment if the issue only occurs in the presence of specific games or while the system is completely idle, how long it takes for the problem to be noticeable, etc.

**Comment by DeathTBO** [comment]
Starting at 20mb, it takes 2 minutes to get to 40mb. It is currently idle, but the problem also happens in every game I've tried (Battlefront 2, Just Cause 2, and Diablo 2).

**Comment by DeathTBO** [comment]
I finished virus scanning, and there were none. I used procmon, and it showed a constant stream of create and exit threads. I also tried rolling back to version 2.8, but that has the same issue.

**Comment by psouza4** [comment]
... 2.8?  Maybe you mean 7.8?

And yes, Borderless Gaming 7.5 - 7.9 will regularly enumerate processes and windows using background threads.  That's at the core of how BG operates and not something that should alarm anyone.

I don't see any memory leaks in the code (and you're welcome to look at it yourself).  It's difficult to leak memory in managed applications anyway since garbage collection is automatic and handled by the .NET framework.  And unfortunately, since you're the only person who seems to be encountering it, tracing what's going on may be impossible.  It's not happening here and nobody else is reporting it, which is what makes me suspicious of your environment.

**Comment by DeathTBO** [comment]
Yes I did mean 7.8. I might try going back to version 6.x.  I can assure you that there are no viruses, and my system is completely up to date.

**Comment by DeathTBO** [comment]
I rolled back to version 6.5 and it appears to be working smoothly. It is bouncing up and down from 18mb-20mb's. I also tested every 7.x version, all had the same results.

**Comment by psouza4** [comment]
Sorry to hear.

We'll leave this issue open in case others have results they can reproduce -- we'll need something to work with to troubleshoot the code, but until we can reproduce the issue here, it'll be nearly impossible to find out if anything's wrong with the code or with your environment.

**Comment by DeathTBO** [comment]
I understand the difficulties. 6.5 works without any problems, and I will test every new version. Hopefully someone else has the same problem (that sounds kind of mean).

**Comment by Azraelle** [comment]
I have memory leak issue too(latest version). After 3 days it ate almost 1gb of ram. And it also consume lik 3-5% of my cpu in iddle state.

**Comment by psouza4** [comment]
Check that you don't have .NET framework corruption.  I've run it for
months without restarting and no issues, as I'm sure others do as well.  So
I would check potential issues with your local system first.

**Comment by JackelTLO** [comment]
I am also having this issue. It starts at 14mb and continues.

**Comment by psouza4** [comment]
Would those of you having issues please try the 8.2 version and see if it has memory issues for you as well?

**Comment by DeathTBO** [comment]
Still having the issue. I downloaded the source code and loaded it into Visual Studio 2015, then started the diagnostic tools. It appears to be an issue with garbage collection.

**Comment by psouza4** [comment]
Doesn't happen here -- I still suspect .NET framework corruption or abnormality.

**Comment by Draygera** [comment]
@DeathTBO What OS are you using? I've noticed that one of the latest .NET security updates for 8.1 has been causing similar issues with certain programs. Reinstalling .NET manually and then reinstalling the updates fixed the problems for me.

**Comment by DeathTBO** [comment]
Windows 7 Professional 64-bit.

**Comment by Draygera** [comment]
@DeathTBO Not quite sure, tbh. But I know that Ubiquitous 2 had this issue but haven't had any problems with this program. Just wanted to throw that out there just in case you were also using 8.1

**Comment by DeathTBO** [comment]
@Draygera Thanks for trying to help. I tried reinstalling my .Net previously, and even more recently I had to install 4.6 for VS 2015.

**Comment by Draygera** [comment]
@DeathTBO Now that I think about it, does Borderless Gaming crash on you? If so, have you checked the Windows Event logs and then copied them over to pastebin.com so that the dev can see what's going on?

**Comment by DeathTBO** [comment]
@Draygera No it doesn't crash, but I haven't left it on for very long. The most ram I let it consume was 200MB, which is when I discovered the issue.

**Comment by nockian** [comment]
I'm running Windows 8.1 and I just checked my task manager processes sorted by memory usage. Yikes!
![borderlessgaming1](https://cloud.githubusercontent.com/assets/16601034/12184987/85784abe-b566-11e5-92eb-8e139badcffa.png)

**Comment by Draygera** [comment]
@nockian What are your specs? O.O

**Comment by nockian** [comment]
@Draygera 
WIndows 8.1 Pro 64-bit
Intel Core i5 3570k
AMD Radeon HD 7970
16G RAM
500G SSD 840

I reinstalled .NET using the repair tool but that didn't seem to fix it; earlier it went back up to 1.5 gigs after I'd left my computer on for a day or so. I read the earlier posts about the .NET security updates, but I have a bunch of those installed and I don't know which one it would be.

**Comment by Firedingo** [comment]
Can Confirm Using Borderless Gaming 8.4 With Windows 7 Professional 64-bit Service Pack 1 and I've seen Borderless gaming use 4.5gb. It seems to have started happening out of nowhere for me.
My system specs are
256gb SSD, 1tb HDD
16gb RAM
Intel Quad Core i7-4720HQ
Nvidia GTX 960m

I'm having a similar issue with another program called DisplayFushion. If anyone has some advice for me I'd appreciate it.

Here's a screenshot,  You can ignore Firefox and Photoshop since I had a large project in Photoshop open for a while and Firefox I had an insane number of tabs open.

![insanememory_usage](https://cloud.githubusercontent.com/assets/15248850/14037552/9158fff2-f299-11e5-9840-49066bf81a2a.png)

**Comment by Firedingo** [comment]
After reading that Bitdefender is known to cause leaking memory issues with DisplayFusion I removed it but quite unexpectedly it was also causing the same issue with Borderless Gaming. In case someone should stumble across this issue in the future I'm mentioning what I discovered in the hopes that it can help others.
It seems in regards to DisplayFusion that BitDefender causes the program to leak memory because of it's scanning function. Here's a link to the issue posted on DisplayFusion's site: [https://www.displayfusion.com/FAQ/#bitdefendermemoryleak](url)

**Comment by DeathTBO** [comment]
I never thought to check BitDefender before. I too use it.
